tensorflow_datasets/object_detection/deeplesion.py

```diff
@@ -0,0 +1,272 @@
+            num_shards=100,
+            gen_kwargs={
+                "lut": lut,
+                "split": train_split,
+            },
+        ),
+        tfds.core.SplitGenerator(
+            name=tfds.Split.VALIDATION,
+            num_shards=10,
+            gen_kwargs={
+                "lut": lut,
+                "split": val_split,
+            },
+        ),
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TEST,
+            num_shards=10,
+            gen_kwargs={
+                "lut": lut,
+                "split": test_split,
+            },
+        ),
+    ]
+
+
+  def _generate_examples(self, lut, split):
+    """Returns examples
+    :type lut: dict, lookup dictionary to lookup whole paths of series_folder
+    :type split: string, path to read ann files
+    :rtype: yield idx, example
+    """
+    for idx, value in enumerate(split.values):
+      fn, bboxs, size, sp = value
+      record = {
+          "image/name": fn,
+          "image": _lookup_image_path(lut, fn),
+          "bbox": _format_bboxs(bboxs, size)
+      }
+      yield idx, record
+
+    
+def _build_lut(paths):
+    """Returns lookup dictionary to lookup whole paths of series_folder
+    :type paths: dict, {<zipfn>:<unzip_path>}
+    :rtype: dict, {<series_folder>:<unzip_path>/Images_png/<series_folder>}
+    """
+    lut = {}
+    for k, v in paths.items():
+        for fd in tf.io.gfile.listdir(os.path.join(v,'Images_png')):
+            lut[fd] = os.path.join(v,'Images_png', fd)
+    return lut
+        
+
+def _lookup_image_path(lut, fn):
+    """Returns whole path of an image
+    """
+    slice_fn = fn.split('_')[-1]
+    fd = '_'.join(fn.split('_')[:-1])
+    
+    return os.path.join(lut[fd], slice_fn)
+    
+    
+def _format_bboxs(bboxs, size):
+    """Return bbox feature
+    :type bboxs: string, "xxx,xxx,xxx,xxx"
+    :type size: string, "xxx,xxx"
+    :rtype: tfds.features.BBox
+    """
+    size = list(map(lambda x: float(x), size.split(',')))
+    coords = list(map(lambda x: float(x), bboxs.split(',')))
+    coords = list(map(lambda x: x if x>0 else 0.0, coords))
+    coords = list(map(lambda x: x if x<size[0] else size[0], coords))
+
+    cnt = int((len(coords))/4)
+
+    bbox_list = []
+    for i in range(cnt):
+        ymin = coords[i*4+1] / size[1]
+        xmin = coords[i*4] / size[0]
+        ymax = coords[i*4+3] / size[1]
+        xmax = coords[i*4+2] / size[0]
+        bbox_list.append(tfds.features.BBox(ymin=ymin, xmin=xmin, ymax=ymax, xmax=xmax))
+    
+    return bbox_list[0]
+    
+    
+class Split:
+    """dataset split builder.
+    """
+    def __init__(self, ann_path):
+        """
+        :type ann_path: string, path of the annotation csv file
+        """
+        self.ann_path = ann_path
+    
+    def split_generator(self):
+        """wrapper of the split
+        """
+        return self._ann_parser()
+    
+    def _ann_parser(self):
+        """parsers the csv file and returns the splits
```

Comment:
please add more explanation on what is inside the CSV file and what you return.

Reply:
> please add more explanation on what is inside the CSV file and what you return.

added it.